zne/noise_amplification/folding_amplifier/glorious_global_folding_amplifier.py

```diff
@@ -86,28 +89,35 @@ def _validate_noise_factor(self, noise_factor: float) -> float:
 
         Raises:
             ValueError: If input noise_factor value is not of type float.
+            TypeError: If input noise_factor value is not of type float.
         """
         try:
             noise_factor = float(noise_factor)
-        except ValueError as exc:
+        except ValueError:
             raise ValueError(
-                f"`noise_factor` is expected to be a positive floating value. "
+                f"`_compute_num_foldings` expects a positive floating value. "
+                f"Received value of {noise_factor} instead."
+            )
+        except TypeError:
+            raise TypeError(
+                f"`_compute_num_foldings` expects a positive floating value. "
                 f"Received value of {noise_factor} instead."
-            ) from exc
+            )
         if noise_factor < 1:
             raise ValueError(
-                f"`noise_factor` is expected to be a positive float noise_factor >= 1."
+                f"`_compute_num_foldings` expects a positive float noise_factor >= 1."
                 f"Received {noise_factor} instead."
             )
         if noise_factor % 2 == 0:
             raise ValueError(
-                f"`noise_factor` is expected to be a positive odd noise_factor. "
+                f"`_compute_num_foldings` expects a positive odd noise_factor. "
```

Comment:
You should not mention private methods like `_compute_num_foldings` in user facing messages (since users should not be aware of those).